azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/features/StorageAccountApi.java

```diff
@@ -0,0 +1,160 @@
+    * POST
+    */
+   @Named("CheckStorageAccountNameAvailability")
+   @POST
+   @Payload("%7B\"name\":\"{name}\",\"type\":\"Microsoft.Storage/storageAccounts\"%7D")
+   @Path("/providers/Microsoft.Storage/checkNameAvailability")
+   Availability isAvailable(@PayloadParam("name") String storageAccountName);
+
+   /**
+    * The Get Storage Account Properties operation returns system properties for the specified storage account.
+    * https://msdn.microsoft.com/en-us/library/mt163553.aspx
+    */
+   @Named("storageaccountproperty:get")
+   @GET
+   @Path("/resourcegroups/{resourceGroup}/providers/Microsoft.Storage/storageAccounts/{storageAccountName}")
+   @Fallback(NullOnNotFoundOr404.class)
+   StorageService get(@PathParam("storageAccountName") String storageAccountName);
+
+   /**
+    * The Get Storage Keys operation returns the primary and secondary access keys for the specified storage account.
+    * https://msdn.microsoft.com/en-us/library/mt163589.aspx
+    * POST
+    */
+   @Named("storageaccountkey:get")
+   @POST
+   @Path("/resourcegroups/{resourceGroup}/providers/Microsoft.Storage/storageAccounts/{storageAccountName}/listKeys")
+   @Fallback(NullOnNotFoundOr404.class)
+   StorageServiceKeys getKeys(@PathParam("storageAccountName") String storageAccountName);
+
+   /**
+    * https://msdn.microsoft.com/en-us/library/mt163567.aspx
+    * POST
+    */
+   @Named("RegenerateStorageAccountKeys")
+   @POST
+   @Payload("%7B\"keyName\":\"{keyName}\"%7D")
+   @Path("/resourcegroups/{resourceGroup}/providers/Microsoft.Storage/storageAccounts/{storageAccount}/regenerateKey")
+   StorageServiceKeys regenerateKeys(@PathParam("storageAccount") String storageAccount,
+                                     @PayloadParam("keyName") String keyName);
+
+   /**
+    * The Update Storage Account asynchronous operation updates the label, the description, and enables or disables the
+    * geo-replication status for the specified storage account. https://msdn.microsoft.com/en-us/library/mt163639.aspx
+    * PATCH
+    */
+   @Named("storageaccount:update")
+   @PATCH
+   @Payload("%7B\"tags\":{tags},\"properties\":{properties}%7D")
+   @MapBinder(BindToJsonPayload.class)
+   @Path("/resourcegroups/{resourceGroup}/providers/Microsoft.Storage/storageAccounts/{storageAccountName}")
+   StorageServiceUpdateParams update(
+           @PathParam("storageAccountName") String storageAccountName,
+           @PayloadParam("properties") StorageServiceUpdateParams.StorageServiceUpdateProperties properties,
+           @PayloadParam("tags") Map<String, String> tags);
+
+   /**
+    * https://msdn.microsoft.com/en-us/library/mt163652.aspx
+    * DELETE
+    */
+   @Named("storageaccount:delete")
+   @DELETE
+   @ResponseParser(StatusCodeParser.class)
+   @Path("/resourcegroups/{resourceGroup}/providers/Microsoft.Storage/storageAccounts/{storageAccountName}")
+   String delete(@PathParam("storageAccountName") String storageAccountName);
```

Comment:
Looking at the docs, it does not seem to return a URL to track an ongoing delete operation; just 200 or 204. WDYT about just returning `void` and remove the response parser? Any other response status code would then throw an exception.